File: omc/android/app/src/main/java/com/pureenee/model/Debt.java
package com.pureenee.model;

import java.util.List;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.relation.ToMany;
import io.objectbox.relation.ToOne;

@Entity
public class Debt {

    @Id
    private long id;

    private double importoDebito;

    private boolean isAttivo;

    ToOne<Ordine> ordine;

    ToMany<SanamentoDebito> sanamenti;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getImportoDebito() {
        return importoDebito;
    }

    public void setImportoDebito(double importoDebito) {
        this.importoDebito = importoDebito;
    }

    public Ordine getOrdine(){
        return this.ordine.getTarget();
    }

    public void setOrdine(Ordine ordine){
        this.ordine.setTarget(ordine);
    }

    public void setOrdine(ToOne<Ordine> ordine) {
        this.ordine = ordine;
    }

    public boolean isAttivo() {
        return isAttivo;
    }

    public void setAttivo(boolean attivo) {
        isAttivo = attivo;
    }

    public List<SanamentoDebito> getSanamenti() {
        return sanamenti;
    }

    public void addSanamento(SanamentoDebito sanamentoDebito) {
        this.sanamenti.add(sanamentoDebito);
    }

    /**
     * Ritorna l'importo tenendo conto dei pagamenti ricevuti
     * @return
     */
    public double getDebitoAttuale(){
        double debitoAttuale = this.importoDebito;
        for (SanamentoDebito sanamento : this.getSanamenti()) {
            debitoAttuale -= sanamento.getImportoSanamento();
        }
        return debitoAttuale;
    }
}

File: deal/lib/model/debt.dart
import 'package:deal/model/sanamento-debt.dart';
import 'package:objectbox/objectbox.dart';

import 'order.dart';

@Entity()
class Debt {

  @Id()
  int id;

  double importoDebito;

  bool isAttivo;

  final order = ToOne<Orders>();

  final sanamenti = ToMany<SanamentoDebt>();

  Debt({
    this.id = 0,
    required this.importoDebito,
    required this.isAttivo
  });

  bool isSanato(){
    return totaleRimastoDaPagare() == 0;
  }

  double totaleRimastoDaPagare(){
    double totaleRimasto = importoDebito;
    sanamenti.forEach(
            (sanamento) => totaleRimasto -= sanamento.importoSanamento!
    );
    return totaleRimasto;
  }


}